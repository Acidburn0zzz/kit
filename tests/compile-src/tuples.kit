function main() {
    var x: (Int, CString, CString) = (1, "second value", "third value");
    var y: (CString, Int) = ("something", 2);
    var z: (CString, Int) = ("something", 2);

    var a: Int;
    var b: CString;
    (a, b, _) = x;
    printf("%i\n", a);
    printf("%s\n", b);
    printf("%s\n", x[2]);

    var c: CString;
    var d: Int;
    (c, d) = ("something", 3);
    printf("%s\n", c);
    printf("%hi\n", d);

    var e = (1, ("inner slot 1", "inner slot 2"));
    var f: Int;
    var g: CString;
    var h: CString;
    (f, (g, h)) = e;
    printf("%hi\n", f);
    printf("%s\n", g);
    printf("%s\n", h);

    // if y == z {
    //     printf("equal\n");
    // }
}
