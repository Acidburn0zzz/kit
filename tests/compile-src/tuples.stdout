1
second value
third value
something
3
1
inner slot 1
inner slot 2
